### PR TITLE
IETF compatible hasher

### PR DESCRIPTION
**Pull request by mmagician**
## Description

closes: https://github.com/w3f/arkworks-algebra/issues/36

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (master)
- [x] Linked to GitHub issue with discussion and accepted design OR have an explanation in the PR that describes this work.
- [x] Wrote unit tests
- [x] Updated relevant documentation in the code
- [ ] Added a relevant changelog entry to the `Pending` section in `CHANGELOG.md`
- [x] Re-reviewed `Files changed` in the GitHub PR explorer

This is just the first part, testing hash-to-field.
The map-to-curve part is still pending.

### CODE REVIEW

File: ec/Cargo.toml [hunk 1]
```diff
@@ -23,11 +23,18 @@ digest = { version = "0.10", default-features = false }
 num-traits = { version = "0.2", default-features = false }
 rayon = { version = "1", optional = true }
 zeroize = { version = "1", default-features = false, features = ["zeroize_derive"] }
+lazy_static = { version = "1.4.0", default-features = false, features = ["spin_no_std"] }
 
 [dev-dependencies]
 hashbrown = { version = "0.11.1" }
-blake2 = { version = "0.10", default-features = false }
-ark-test-curves = { version = "^0.3.0", path = "../test-curves", default-features = false, features = ["bls12_381_curve"] }
+ark-test-curves = { version = "^0.3.0", path = "../test-curves", default-features = false, features = [ "bls12_381_curve"] }
```
Comment (drskalman) [inline ec/Cargo.toml:30]:
```suggestion
ark-test-curves = { version = "^0.3.0", path = "../test-curves", default-features = false, features = ["bls12_381_curve"] }
```

**Comment by mmagician** [comment]
@drskalman [this PR](https://github.com/arkworks-rs/curves/pull/100) should hopefully remove the outstanding CI jobs failures (even though it was not related to our code).

**Comment by drskalman** [comment]
I moved eb2a8a9 to ietf-map-to-curve-tester branch.

**Comment by drskalman** [comment]
> @drskalman [this PR](https://github.com/arkworks-rs/curves/pull/100) should hopefully remove the outstanding CI jobs failures (even though it was not related to our code).

Why don't you merge it into the pull request then?

**Comment by mmagician** [comment]
> Why don't you merge it into the pull request then?

Not sure I follow. The PR is already merged on arkworks master (curves).
Since the CI job pulls the latest curves, it was failing since there was a bug in which dependencies curves were using.

**Pull request closed without merging**